lib/developer_portal/lib/liquid/drops/message.rb

```diff
@@ -31,7 +31,8 @@ def created_at
       desc "URL of the message detail, points either to inbox or outbox."
       def url
         if @model.hidden_at.present?
-          admin_messages_trash_path(id)
+          # URLs in Trash always use Message (not MessageRecipient) ID
+          admin_messages_trash_path(message.id)
```

Comment:
provided we didn't change this, it's fine. But in general, doesn't it make more sense to show recipient id in trash as well? After all a user would see only its "recipients", no?

Reply:
Yes, it could make more sense (the info would be the same, but the user would be able to see whether the message is read or not), but it's a bit more tricky to implement, because we'd need to somehow fetch the different kinds of associations separately, which is not hard as such, but is somewhat more complicated when there is pagination.

Or, we would then need to have two different controller actions - one for each type.
I don't know... when I looked at it, this was the easiest way to implement.

Also, it's the same thing trash is currently handled in the admin portal.

Reply:
about fetching, we need to just use the `with_message` scope of `MessageReceiver`. Actually if we use the `received_messages` scope  of account, then that should already be there. Anyway, just a suggestion.

Reply:
Actually the logical thing to do is use message recipient when available and message - when not. If we are the sender, then we don't have  a recipient for ourselves. So have to use the message. If we are the receiver, then we have a message recipient and should use that instead.

I'll merge and leave further refinement for a later PR.